src/gui/uithememanager.cpp

```diff
@@ -30,17 +30,489 @@
+        {"TrackerList.UpAction", "go-up"},
+        {"TransferFilter.AllTrackers", "network-server"},
+        {"TransferFilter.Tracker", "network-server"},
+        {"TransferFilter.Tracker.PauseAction", "media-playback-pause"},
+        {"TransferFilter.Tracker.RemoveAction", "edit-delete"},
+        {"TransferFilter.Tracker.ResumeAction", "media-playback-start"},
+        {"TransferFilter.WithoutTrackers", "network-server"},
+        {"TransferList.BottomAction", "go-bottom"},
+        {"TransferList.Category", "view-categories"},
+        {"TransferList.Category.AddAction", "list-add"},
+        {"TransferList.Category.ResetAction", "edit-clear"},
+        {"TransferList.Category.SavePath", "inode-directory"},
+        {"TransferList.CopyAction", "edit-copy"},
+        {"TransferList.DestinationFolder.OpenAction", "inode-directory"},
+        {"TransferList.DownAction", "go-down"},
+        {"TransferList.DownloadLimit", "kt-set-max-download-speed"},
+        {"TransferList.ForceReannounce", "document-edit-verify"},
+        {"TransferList.ForceRecheck", "document-edit-verify"},
+        {"TransferList.ForceResumeAction", "media-seek-forward"},
+        {"TransferList.Hash.CopyAction", "edit-copy"},
+        {"TransferList.MagnetLink.CopyAction", "kt-magnet"},
+        {"TransferList.MaxRatio", "ratio"},
+        {"TransferList.Name.CopyActin", "edit-copy"},
+        {"TransferList.PauseAction", "media-playback-pause"},
+        {"TransferList.PreviewAction", "view-preview"},
+        {"TransferList.RemoveAction", "edit-delete"},
+        {"TransferList.RenameAction", "edit-rename"},
+        {"TransferList.ResumeAction", "media-playback-start"},
+        {"TransferList.Tackers.RenameAction", "edit-rename"},
+        {"TransferList.Tags", "view-categories"},
+        {"TransferList.Tags.AddAction", "list-add"},
+        {"TransferList.Tags.RemoveAllAction", "edit-clear"},
+        {"TransferList.TopAction", "go-top"},
+        {"TransferList.TorrentSaveLocation", "inode-directory"},
+        {"TransferList.UpAction", "go-up"},
+        {"TransferList.UploadLimit", "kt-set-max-upload-speed"},
+        {"Tray.DarkLogo", "qbittorrent-tray-dark"}, {"Tray.Download", "download"},
+        {"Tray.LightLogo", "qbittorrent-tray-light"},
+        {"Tray.Logo", "qbittorrent-tray"}, {"Tray.Seeding", "seeding"},
+        {"WebSeed.AddAction", "list-add"}, {"WebSeed.RemoveAction", "list-remove"},
+        {"WebSeedURL.CopyAction", "edit-copy"},
+        {"WebSeedURL.EditAction", "edit-rename"} };
+#endif
+}
 
 UIThemeManager *UIThemeManager::m_instance = nullptr;
 
+namespace
+{
+    class ThemeError : public RuntimeError
+    {
+    public:
+        using RuntimeError::RuntimeError;
+    };
+
+    /*
+        Loads Json Icon Config file from `configFile`, and transforms it into a map
+
+        function throws ThemeError if encountered with any error
+     */
+    QHash<QString, QString> loadIconConfig(const QString &configFile)
```

Comment:
It should be a static member since It doesn't require anything from `uithememanager`.

Reply:
Sorry. I mean different things.
If we really don't need different icon config files then you can change it to non static member function that don't return its result but just modify corresponding class fields. Otherwise leave it as is (resolving other comments).

Reply:
should I do it inplace since there is only one iconconfig?

Reply:
Seems it degrades the readability of the calling code.
Splitting into sub-functions is still useful in terms of code structuring. If some code portion does "stand-alone" job, it makes sense to put it in a separate method (unless it's really trivial).

Reply:
> should I do it inplace since there is only one iconconfig?

I decided against it since this makes constructor too big.